app/views/test/components/common/modal/block.html.haml

```diff
@@ -0,0 +1,3 @@
+= render ReactComponents::Common::Modal.new do
+  %h1 I am some code in a block
+  %button{ data: { modal_close: true } } Close
```

Comment:
@kntsoriano FYI, I added two ways to use the modal. This is a slightly more elegant refactoring of your original way (render the contents of the block, rather than capture and then render as seperate stages)